test/action_controller/json/include_test.rb

```diff
@@ -114,6 +126,54 @@ def test_render_resource_with_deep_include
           assert_equal(expected_deep_include_response, response)
         end
 
+        def test_render_resource_with_wildcard_include
+          get :render_resource_with_wildcard_include
+
+          response = JSON.parse(@response.body)
+
+          assert_equal(expected_wildcard_include_response, response)
+        end
+
+        def test_render_resource_with_wildcard_include_with_wildcards_disabled
+          with_wildcards_disabled do
+            get :render_resource_with_wildcard_include
+          end
+
+          expected = {
+            "author" => {
+              "id" => 1,
+              "name" => "Steve K."
+            }
+          }
+
+          response = JSON.parse(@response.body)
+
+          assert_equal(expected, response)
+        end
+
+        def test_render_resource_with_deep_wildcard_include
+          assert_raise SystemStackError do
+            get :render_resource_with_deep_wildcard_include
```

Comment:
`**` wildcard would cause stack overflow in this case. I guess this test can be removed, but left it here as a PoC for now.